### ISSUE TITLE
MacOS Bug

### ISSUE THREAD

**Issue by jiatao99** [m4rs-mt/ILGPU#334]
I tried to use your wrapper in MacOS and found a couple bugs:

1. `[DllImport("libOpenCL", EntryPoint = "xxx")]` is incorrect. The name should be OpenCL. 
      - I copied OpenCL to libOpenCL in Mac
      - I add `LD_LIBRARY_PATH` in the environment variable
      - the CurrentApi.IsSupported now becomes true 
2. I got AMD Radeon 5300M OpenCL id using something like below
```
	CLAcceleratorId cLAcceleratorId = new CLAcceleratorId(intPtr, intPtr2);
         if (cLAcceleratorId.DeviceType == CLDeviceType.CL_DEVICE_TYPE_GPU)
         {
                  var deviceName = CLAPI.CurrentAPI.GetDeviceInfo(cLAcceleratorId.DeviceId, CLDeviceInfoType.CL_DEVICE_NAME);
                  if (deviceName.StartsWith("AMD"))
                            return cLAcceleratorId;
```
3. When I tried to create Accelerator `Accelerator.Create(context, amdClID)`, I got this error: 
    - 'System.EntryPointNotFoundException' occurred in ILGPU.dll: 'Unable to find an entry point named 'clCreateCommandQueueWithProperties'

Any idea how to bypass this error?

**Comment by m4rs-mt** [comment]
@jiatao99 Thank you for submitting this issue report. We already revised the `OpenCL` library name on MacOS based on some feedback in one of our latest releases. However, as far as I know Apple has [discontinued support for OpenCL](https://support.apple.com/en-us/HT202823) and supports OpenCL 1.2 only. The error message you are referring to is definitively a bug, since the error message should be more meaningful 😄

Since ILGPU requires OpenCL >= 2.0, there is no workaround for this issue on Apple systems that I am aware of at the moment.

**Comment by jiatao99** [comment]
I found the bug in CLAPI.cs

in CreateCommandQueue function, instead of clCreateCommandQueue you called clCreateCommandQueueWithProperties which causing the exception thrown in MacOS. I recompiled your project and this error is gone

However, I do get NotSupportException accelerator.LoadAutoGroupedStreamKernel indicating OpenCL >= 2.0 is required

I guess this is a dead end for MacOS.

**Comment by m4rs-mt** [comment]
Thanks for digging into this 🤓. However, as I explained above by providing the link, `clCreateCommandQueueWithProperties` is available in OpenCL >= 2.0 and offers much better interop with the OpenCL platform. Therefore, I argue that this is not the actual "bug" you are referring to. Instead, we should throw a better error message, because the driver will not be compatible in any way.

**Comment by jiatao99** [comment]
Agree. For MacOS, everything is stayed at version 1.2. 

For Linux, the AMD driver only support C 1.2. I will try ROCm driver next. Hopefully that works. 

Do you have instruction on how to make it work on Linux (assume it is C 2.0)? Finding libOpenCL in Linux is very tricky. I still haven't figure out how.
